v1.7.0: configure fails with `--enable-debug`

Hi,

Configure fails on our CentOS 6 system when using the `--enable-debug` with the error
```
configure: error: dlopen not found.  libfabric requires libdl.
```
This is because it has ended up with `-fstack-protector-strong` in `CFLAGS` even though it's not supported by our GCC version; all tests after this point the fail until it reaches the libdl test which is fatal.

It looks like it does test for `-fstack-protector-strong` and correctly detect that it's unsupported, but then leaves that flag in `CFLAGS`. I think commit 7bb6649ad6fb7bedaf5d387d68e4266e0d00feed is the culprit here; I'll patch it and PR.

Cheers,
Ben

--- a/configure.ac
+++ b/configure.ac
@@ -97,9 +97,10 @@ AS_IF([test x"$enable_debug" != x"no"],
       [dbg=1
        # See if all the flags in $debug_c_other_flags work
        good_flags=
+       CFLAGS_save="$CFLAGS"
        for flag in $debug_c_other_flags; do
            AC_MSG_CHECKING([to see if compiler supports $flag])
-           CFLAGS="$flag $CFLAGS"
+           CFLAGS="$flag $CFLAGS_save"
            AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[int i = 3;]])],
 	                     [AC_MSG_RESULT([yes])
 			      good_flags="$flag $good_flags"],
@@ -108,7 +109,8 @@ AS_IF([test x"$enable_debug" != x"no"],
        debug_c_other_flags=$good_flags
        unset good_flags
 
-       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} $CFLAGS"],
+       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} ${CFLAGS_save}"
+       unset CFLAGS_save],
       [dbg=0
        CFLAGS="-O2 -DNDEBUG $CFLAGS"])
 

